docs/spark/tutorials/ml-sentment-analysis.md

```diff
@@ -0,0 +1,207 @@
+
+Add the following code to your `Main` method to register a UDF called `MLudf`.
+
+```csharp
+spark.Udf()
+    .Register<string, bool>("MLudf", predict);
+```
+
+This UDF takes a Yelp review string as input, and outputs true or false for positive or negative sentiments, respectively. It uses the *predict()* method that you define in a later step.
+
+### Use Spark SQL to call the UDF
+
+Now that you've read in your data and incorporated ML, use Spark SQL to call the UDF that will run sentiment analysis on each row of your DataFrame. Add the following code to your `Main` method:
+
+```csharp
+// Use Spark SQL to call ML.NET UDF
+// Display results of sentiment analysis on reviews
+df.CreateOrReplaceTempView("Reviews");
+DataFrame sqlDf = spark.Sql("SELECT ReviewText, MLudf(ReviewText) FROM Reviews");
+sqlDf.Show();
+
+// Print out first 20 rows of data
+// Prevent data getting cut off by setting truncate = 0
+sqlDf.Show(20, 0, false);
+
+spark.Stop();
+```
+
+### Create predict() method
+
+Add the following code before your `Main()` method. This code is similar to what is produced by Model Builder in *ConsumeModel.cs*. Moving this method to your console keeps the model loading every time you run your app.
+
+```csharp
+private static readonly PredictionEngine<ModelInput, ModelOutput> _predictionEngine;
+
+static Program()
+{
+    MLContext mlContext = new MLContext();
+    ITransformer model = mlContext.Model.Load("MLModel.zip", out DataViewSchema schema);
+    _predictionEngine = mlContext.Model.CreatePredictionEngine<ModelInput, ModelOutput>(model);
+}
+
+static bool predict(string text)
+{
+    ModelInput input = new ModelInput
+    {
+        ReviewText = text
+    };
+
+    return _predictionEngine.Predict(input).Prediction;
+}
+```
+
+## Add the model to your console app
+
+In Solution Explorer, copy the *MLModel.zip* file from the **MLSparkModelML.Model** project and paste it in the **MLSparkModelML.ConsoleApp** project. A reference is automatically added in *MLSparkModelML.ConsoleApp.csproj*.
+
+## Run your code
+
+Use `spark-submit` to run your code. Navigate to your console app's root folder using the command prompt and run the following commands.
+
+First, clean and publish your app.
+
+```dotnetcli
+dotnet clean
+dotnet publish
+```
+
+Then navigate to the console app's publish folder and run the following `spark-submit` command. Remember to update the command with the actual path of your Microsoft Spark jar file.
+
+```dotnetcli
+%SPARK_HOME%\bin\spark-submit --class org.apache.spark.deploy.dotnet.DotnetRunner --master local microsoft-spark-2.4.x-0.10.0.jar dotnet MLSparkModelML.ConsoleApp.dll
+```
+
+## Get the code
+
+This tutorial uses code from the [Sentiment Analysis with Big Data](https://github.com/dotnet/spark/tree/master/examples/Microsoft.Spark.CSharp.Examples/MachineLearning/Sentiment) example.
```

Comment:
I looked at the code in this folder, and it doesn't match the code in the article. Was this adapting an earlier sample? Or is the code to match in another PR?

Reply:
No, I adapted this code based on Model Builder output. I changed the wording to note that it's similar - not identical.